cmd/utils.go

```diff
@@ -26,6 +29,14 @@ var (
 	cache = storage.NewStorage()
 )
 
+// CastDNSEntry is used by DNS and caching discovery
+type CastDNSEntry interface {
```

Comment:
Can this be moved to `discovery` package?

Reply:
It is useful only in this file: I unexported it to make this clearer.